Mathlib/MeasureTheory/Measure/MeasureSpace.lean

```diff
@@ -0,0 +1,4677 @@
+  exact fun ⟨i, hi⟩ => ⟨i + (m - n), by simpa only [add_assoc, tsub_add_cancel_of_le hnm] using hi⟩
+#align measure_theory.measure_limsup_eq_zero MeasureTheory.measure_limsup_eq_zero
+
+theorem measure_liminf_eq_zero {s : ℕ → Set α} (h : (∑' i, μ (s i)) ≠ ⊤) :
+    μ (liminf s atTop) = 0 := by
+  rw [← le_zero_iff]
+  have : liminf s atTop ≤ limsup s atTop :=
+    liminf_le_limsup
+      (by isBoundedDefault)
+      (by isBoundedDefault)
+  exact (μ.mono this).trans (by simp [measure_limsup_eq_zero h])
+#align measure_theory.measure_liminf_eq_zero MeasureTheory.measure_liminf_eq_zero
+
+theorem limsup_ae_eq_of_forall_ae_eq (s : ℕ → Set α) {t : Set α}
+    (h : ∀ n, s n =ᵐ[μ] t) : @limsup (Set α) ℕ _ s atTop =ᵐ[μ] t := by
+    -- Need `@` below because of diamond; see gh issue #16932
+  simp_rw [ae_eq_set] at h⊢
+  constructor
+  · rw [atTop.limsup_sdiff s t]
+    apply measure_limsup_eq_zero
+    simp [h]
+  · rw [atTop.sdiff_limsup s t]
+    apply measure_liminf_eq_zero
+    simp [h]
+#align measure_theory.limsup_ae_eq_of_forall_ae_eq MeasureTheory.limsup_ae_eq_of_forall_ae_eq
+
+theorem liminf_ae_eq_of_forall_ae_eq (s : ℕ → Set α) {t : Set α}
+    (h : ∀ n, s n =ᵐ[μ] t) : @liminf (Set α) ℕ _ s atTop =ᵐ[μ] t := by
+    -- Need `@` below because of diamond; see gh issue #16932
+  simp_rw [ae_eq_set] at h⊢
+  constructor
+  · rw [atTop.liminf_sdiff s t]
+    apply measure_liminf_eq_zero
+    simp [h]
+  · rw [atTop.sdiff_liminf s t]
+    apply measure_limsup_eq_zero
+    simp [h]
+#align measure_theory.liminf_ae_eq_of_forall_ae_eq MeasureTheory.liminf_ae_eq_of_forall_ae_eq
+
+theorem measure_if {x : β} {t : Set β} {s : Set α} :
+    μ (if x ∈ t then s else ∅) = indicator t (fun _ => μ s) x := by split_ifs with h <;> simp [h]
+#align measure_theory.measure_if MeasureTheory.measure_if
+
+end
+
+section OuterMeasure
+
+variable [ms : MeasurableSpace α] {s t : Set α}
+
+/-- Obtain a measure by giving an outer measure where all sets in the σ-algebra are
+  Carathéodory measurable. -/
+def OuterMeasure.toMeasure (m : OuterMeasure α) (h : ms ≤ m.caratheodory) : Measure α :=
+  Measure.ofMeasurable (fun s _ => m s) m.empty fun _f hf hd =>
+    m.unionᵢ_eq_of_caratheodory (fun i => h _ (hf i)) hd
+#align measure_theory.outer_measure.to_measure MeasureTheory.OuterMeasure.toMeasure
+
+theorem le_toOuterMeasure_caratheodory (μ : Measure α) : ms ≤ μ.toOuterMeasure.caratheodory :=
+  fun _s hs _t => (measure_inter_add_diff _ hs).symm
+#align measure_theory.le_to_outer_measure_caratheodory MeasureTheory.le_toOuterMeasure_caratheodory
+
+@[simp]
+theorem toMeasure_toOuterMeasure (m : OuterMeasure α) (h : ms ≤ m.caratheodory) :
+    (m.toMeasure h).toOuterMeasure = m.trim :=
+  rfl
+#align measure_theory.to_measure_to_outer_measure MeasureTheory.toMeasure_toOuterMeasure
+
+-- Porting note: A coercion is directly elaborated in Lean4, so the LHS is simplified by
+-- `toMeasure_toOuterMeasure` even if this theorem has high priority.
+@[simp, nolint simpNF]
```

Comment:
For all of these `nolint simpNF` lemmas, could you add a new version of the lemma with the `simp` attribute, the same RHS as this lemma, and the LHS in simp-normal form

Reply:
The theorem correspending `MeasureTheory.Measure.sum_coe_finset` couldn't be simplified.